tiledb/sm/array/array_directory.h

```diff
@@ -126,6 +126,27 @@ class ArrayDirectory {
       return fragment_uris_;
     };
 
+    /** Returns the filtered fragment URIs matching the given fragments_list. */
+    std::vector<TimestampedURI> fragment_uris(
+        std::vector<std::string> fragments_list) const {
+      std::vector<TimestampedURI> uris;
+      for (auto fragment : fragments_list) {
+        for (auto uri : fragment_uris_) {
+          if (URI(fragment) == uri.uri_) {
+            uris.emplace_back(uri);
+            break;
+          } else {
+            if (uri == fragment_uris_.back()) {
```

Comment:
This is throwing if a user is trying to retrieve from `fragment_uris_` a fragment that does not exist on the `ArrayDirectory`. We _could_ ignore such behavior, but since this is intended as a helper function to delete fragments by name, I believe it's best to disallow trying to delete fragment that doesn't exist.